client.go

```diff
@@ -38,6 +39,10 @@ var (
 	// defaultClient is used for performing requests without explicitly making
 	// a new client. It is purposely private to avoid modifications.
 	defaultClient = NewClient()
+
+	// We need to consume response bodied to maintain http connections, but
+	// limit the size we consume to respReadLimit.
+	respReadLimit = int64(4 << 10)
```

Comment:
I would probably just specify 4096 here for clarity, but a comment in-line would also suffice.